src/NodeValueReader.cpp

```diff
@@ -0,0 +1,487 @@
+{
+  //std::cout << indent(-1) << "NodeValueReader::end_element()" << std::endl;
+  ++current_index_;
+  return true;
+}
+
+bool NodeValueReader::read_boolean(ACE_CDR::Boolean& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_boolean()" << std::endl;
+  return primitive_helper<v8::Boolean>(value, &v8::Value::IsBoolean);
+}
+
+bool NodeValueReader::read_byte(ACE_CDR::Octet& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_byte()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtoul);
+}
+
+#if OPENDDS_HAS_EXPLICIT_INTS
+
+bool NodeValueReader::read_int8(ACE_CDR::Int8& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_int8()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtol);
+}
+
+bool NodeValueReader::read_uint8(ACE_CDR::UInt8& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_uint8()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtoul);
+}
+
+#endif
+
+bool NodeValueReader::read_int16(ACE_CDR::Short& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_int16()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtol);
+}
+
+bool NodeValueReader::read_uint16(ACE_CDR::UShort& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_uint16()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtoul);
+}
+
+bool NodeValueReader::read_int32(ACE_CDR::Long& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_int32()" << std::endl;
+  bool result = primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtol);
+  if (result) {
+    //std::cout << indent(0) << " - found value '" << value << "'" << std::endl;
+  }
+  return result;
+}
+
+bool NodeValueReader::read_uint32(ACE_CDR::ULong& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_uint32()" << std::endl;
+  return primitive_helper<v8::Integer>(value, &v8::Value::IsNumber, strtoul);
+}
+
+bool NodeValueReader::read_int64(ACE_CDR::LongLong& value)
+{
+  //std::cout << indent(0) << "NodeValueReader::read_int64()" << std::endl;
+  Nan::MaybeLocal<v8::Value> mlvai = use_name_ ? Nan::Get(current_object_, current_property_name_) : Nan::Get(current_object_, current_index_);
+  if (!mlvai.IsEmpty()) {
+    // This doesn't work because NaN seems to lack conversion support to BigInt
+    // Not super surprising, since BigInt support wasn't introduced until Node.js 10.4
+    //{
+    //  Nan::MaybeLocal<v8::BigInt> tov = Nan::To<v8::BigInt>(mlvai.ToLocalChecked());
+    //  if (!tov.IsEmpty()) {
+    //    value = static_cast<ACE_CDR::LongLong>(tov.ToLocalChecked()->Int64Value());
+    //    return true;
+    //  }
+    //}
+    {
+      Nan::MaybeLocal<v8::String> tov = Nan::To<v8::String>(mlvai.ToLocalChecked());
+      if (!tov.IsEmpty()) {
+        std::string temp;
+        read_string(temp);
+#ifndef ACE_LINUX
```

Comment:
Is there a more specific macro we could use?  This doesn't seem to be related to the OS (for example, different compilers on the same OS may do this differently).